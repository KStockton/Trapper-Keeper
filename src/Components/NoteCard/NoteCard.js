import React, { Component } from "react";
import NoteItem from "../NoteItem/NoteItem";

class NoteCard extends Component {
  state = { title: "" };

  handleChange = e => {
    const { name, value } = e.target;
    this.setState({ [name]: value });
  };

  handleSubmit = () => {
    console.log('YO')
  };

  render() {
    console.log('notecard')
    return (
      <div className="note-card-component">
        <section className="note-card">
          <input
            type="text"
            className="note-title"
            value={this.state.title}
            onChange={this.handleChange}
            placeholder="Title"
            name="title"
          />
          <NoteItem />
          <input
            type="submit"
            className="btn"
            onClick={() => this.handleSubmit()}
            value="Save"
          />
        </section>
      </div>
    );
  }
}

export default NoteCard;
