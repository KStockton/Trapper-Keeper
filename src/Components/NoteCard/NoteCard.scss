.note-card {
  background-color: #ffff88;
  height: 200px;
  width: 200px;
  border-radius: 5%;

  .note-title {
    margin-top: 15px;
    background-color: #ffff88;
    border: none;
    width: 99%;
    height: 1em;
    font-size: 35px;
    padding-bottom: 15px;
  }
}