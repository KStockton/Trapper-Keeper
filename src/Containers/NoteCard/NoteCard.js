import React, { Component } from "react";
import NoteItem from "../../Components/NoteItem/NoteItem";
import { fetchDeleteNote } from "../../Api/fetch/fetchDeleteNote";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import * as actions from "../../Actions/index";

export class NoteCard extends Component {
  constructor() {
    super();
    this.state = { title: "", delete: false };
  }

  renderListItems = () => {
    const { notes } = this.props.data;
    return notes.map(task => <NoteItem key={task._id} tasks={task} />);
  };

  handleMouseOver = () => {
    this.setState({ delete: !this.state.delete });
  };

  handleDelete = async noteId => {
    try {
      await fetchDeleteNote(noteId);
      this.props.deleteNote(noteId);
    } catch (error) {
      this.props.handleError(error.message);
    }
  };

  render() {
    const { title, _id } = this.props.data;
    return (
      <div className="note-card-component">
        <div className="note-title">{title}</div>
        {this.renderListItems()}
        <section className="note-options">
          <Link to={`notes/${_id}`}>
            <button className="edit-note-btn">Edit Note</button>
          </Link>
          {this.state.delete === true ? (
            <div
              id="deletebtn"
              className="red-delete-btn"
              onMouseOver={this.handleMouseOver}
              onMouseLeave={this.handleMouseOver}
              onClick={() => this.handleDelete(_id)}
            />
          ) : (
            <div className="delete-btn" onMouseOver={this.handleMouseOver} />
          )}
        </section>
      </div>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  deleteNote: id => dispatch(actions.deleteNote(id)),
  handleError: errorMessage => dispatch(actions.handleError(errorMessage))
});

export default connect(
  null,
  mapDispatchToProps
)(NoteCard);
