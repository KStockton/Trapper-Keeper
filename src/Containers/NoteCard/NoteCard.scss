.note-card {
  background-color: #ffff88;
  box-shadow: 3px 10px 10px -10px rgba(0,0,0,0.75);
  width: 100%;
  margin: 0 0 20px;
  padding: 10px;
  overflow: hidden;
  break-inside: avoid;
  text-decoration: none !important;

  .note-title {
    margin-top: 15px;
    background-color: #ffff88;
    border: none;
    width: 99%;
    height: 1em;
    font-size: 35px;
    padding-bottom: 15px;
    text-decoration: none !important;
  }

  .btn {
    cursor: pointer;
  }

  .delete-btn {
    background-image: url('../Assets/trashcan.svg');
    background-size: contain;
    background-repeat: no-repeat;
    height: 30px;
    width: 30px;
    cursor: pointer;
  }

  .red-delete-btn {
    background-image: url('../Assets/red-waste-bin.svg');
    background-size: contain;
    background-repeat: no-repeat;
    height: 30px;
    width: 30px;
    cursor: pointer;
  }

  .note-options {
    display: flex;
    flex-direction: row;
    justify-content: space-around;
  }
}