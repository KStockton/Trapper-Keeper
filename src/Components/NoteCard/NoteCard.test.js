import React from "react";
import { shallow } from "enzyme";
import NoteCard from "./NoteCard";

describe("NoteCard", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<NoteCard />);
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
