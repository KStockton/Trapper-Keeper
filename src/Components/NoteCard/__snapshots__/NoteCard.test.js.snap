// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteCard should match the snapshot 1`] = `undefined`;
